lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -71,26 +168,154 @@ def edit_subscriber(self, subscriber_id):
+        """
+        with self.conn:
+            res = self.conn.execute(
+                "SELECT data FROM subscriberdb WHERE " "subscriber_id = ?",
+                (sid,),
+            )
+            row = res.fetchone()
+            if not row:
+                raise SubscriberNotFoundError(sid)
+            non_3gpp = Non3GPPUserProfile()
+            sub_data = self.get_subscriber_data(sid)
+            num_stored_apn = len(sub_data.non_3gpp.apn_config)
+            num_rcvd_apn = len(request.non_3gpp.apn_config)
+            for idx in range(num_rcvd_apn):
+                for itrn in range(num_stored_apn):
+                    if (
+                        sub_data.non_3gpp.apn_config[itrn].service_selection
+                        == request.non_3gpp.apn_config[idx].service_selection
+                    ):
+                        del sub_data.non_3gpp.apn_config[itrn]
+                        # Delete the sid entry from apndb
+                        try:
+                            self._delete_sid_from_apndb(
+                                request.non_3gpp.apn_config[
+                                    idx
+                                ].service_selection,
+                                sid,
+                            )
+
+                        except ApnNotFoundError as e:
+                            logging.warning(
+                                "sid entry not found for APN : %s %s",
+                                e,
+                                request.non_3gpp.apn_config[
+                                    0
+                                ].service_selection,
+                            )
+
+                            break
+
+            data_str = sub_data.SerializeToString()
+            # Re-populate subscriber data with the APN parameters
+            self.conn.execute(
+                "UPDATE subscriberdb SET data = ? " "WHERE subscriber_id = ?",
+                (data_str, sid),
+            )
 
     def delete_subscriber(self, subscriber_id):
         """
         Method that deletes a subscriber, if present.
         """
         with self.conn:
-            self.conn.execute("DELETE FROM subscriberdb WHERE "
-                              "subscriber_id = ?", (subscriber_id, ))
+            # Delete the subscriber id from the apndb
+            try:
+                sub_data = self.get_subscriber_data(subscriber_id)
+            except SubscriberNotFoundError:
+                raise SubscriberNotFoundError()
+            num_apn = len(sub_data.non_3gpp.apn_config)
+            for idx in range(num_apn):
+                try:
+                    self._delete_sid_from_apndb(
+                        sub_data.non_3gpp.apn_config[idx].service_selection,
+                        subscriber_id,
+                    )
+                except ApnNotFoundError as e:
+                    logging.warning(
+                        "sid entry not found for APN : %s %s",
```

Comment:
From Tariq: "
Not very accurate. Its more of "APN not found".
Actually we dont raise error when SID is not associated with a valid APN
"

Reply:
_delete_sid_from_apndb is a common function and the exception will be valid in some cases. Hence delete_subscriber function will just ignore the exception and continue to delete the subscriber